plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastoreModule.java

```diff
@@ -55,6 +57,10 @@ public HiveMetastore createCachingHiveMetastore(
         HiveMetastore decoratedDelegate = hiveMetastoreDecorator
                 .map(decorator -> decorator.decorate(delegate))
                 .orElse(delegate);
+        if (!nodeManager.getCurrentNode().isCoordinator()) {
+            // disable caching on workers to avoid bugs due to inconsistent cache between worker and coordinator
+            return decoratedDelegate;
```

Comment:
```
// Disable caching on workers, because there currently is no way to invalidate such a cache.
// Note: while we could skip `CachingHiveMetastoreModule` altogether on workers, we retain it so that catalog configuration
// can remain identical for all nodes, making cluster configuration easier.
```